programs/lz4cli.c

```diff
@@ -399,6 +402,21 @@ int main(int argc, const char** argv)
                     /* Compression (default) */
                 case 'z': mode = om_compress; break;
 
+                case 'D':
+                    if (argument[1] == '\0') {
+                        /* path is next arg */
+                        if (i + 1 == argc) {
+                            /* there is no next arg */
+                            badusage(exeName);
+                        }
+                        dictionary_filename = argv[++i];
+                    } else {
+                        /* path follows immediately */
+                        dictionary_filename = argument + 1;
+                    }
+                    break_inner_loop = 1;
```

Comment:
For the same effect, you could also move `argument` to the end of the string, using : 
`argument += strlen(argument) - 1;`
Outcome is the same (breaking inner loop, moving to next command line argument).
Difference is, it avoids introducing another mutable variable, which is one more state to track during debugging.

*Note* : `- 1` because `lz4cli` uses the convention of letting the main loop move argument forward by 1 character. This is is not a great choice btw : it has been changed in `zstdcli`. One day, it will be useful for both cli to use the same convention.